File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0, r = 0, res=0;

        while(l<=r && r < prices.size()) {
            if(prices[r] > prices[l]) {
                res = max(res, prices[r]-prices[l]);
            } else {
                l = r;
            }
            r+=1;
        }

        return max(res, 0);
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        currPrice = prices[0]
        res = 0

        for i in prices:
            if i < currPrice:
                currPrice = i
            else:
                res = max(res, i - currPrice)
        
        return res